cmd/auth/auth.go

```diff
@@ -0,0 +1,200 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"path"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+const (
+	codeChallengeLength = 87
+	supabaseID          = "ibcwmlhcimymasokhgvn"
+	supabasePublicKey   = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYW5vbiIsImlhdCI6MTYyOTkzMDc3OCwiZXhwIjoxOTQ1NTA2Nzc4fQ.zcdbd7kDhk7iNSMo8SjsTaXi0wlLNNQcSZkzZ84NUDg"
+	authCallbackAddr    = "localhost:3000"
+	sessionFileName     = "session.json"
+)
+
+//go:embed success.html
+var successHTML string
+
+func NewLoginCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "Log into the CLI application via GitHub",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run()
+		},
+	}
+
+	return cmd
+}
+
+func run() error {
+	codeVerifier, codeChallenge, err := pkce(codeChallengeLength)
+	if err != nil {
+		return fmt.Errorf("PKCE error: %v", err)
+	}
+
+	supabaseAuthURL := fmt.Sprintf("https://%s.supabase.co/auth/v1/authorize", supabaseID)
+	queryParams := url.Values{
+		"provider":              {"github"},
+		"code_challenge":        {codeChallenge},
+		"code_challenge_method": {"S256"},
+		"redirect_to":           {"http://" + authCallbackAddr + "/"},
+	}
+
+	authenticationURL := supabaseAuthURL + "?" + queryParams.Encode()
+
+	server := &http.Server{Addr: authCallbackAddr}
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		defer shutdown(server)
+
+		code := r.URL.Query().Get("code")
+		if code == "" {
+			http.Error(w, "Error: auth-code query param not found", http.StatusBadRequest)
+			return
+		}
+
+		sessionData, err := getSession(code, codeVerifier)
+		if err != nil {
+			http.Error(w, "Error: Access token exchange failed", http.StatusInternalServerError)
+			return
+		}
+
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Println("Error getting home directory:", err)
```

Comment:
This should be `fmt.Errorf` with the error attached.

Reply:
https://github.com/open-sauced/pizza-cli/commit/5a0b17de099522911138dd11169abf3c22888b22